visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -371,12 +375,53 @@ public void sauceVisualCheck(String snapshotName, CheckOptions options) {
     }
   }
 
+  private DiffingOptionsIn.Builder setDiffingOptionValue(
+      DiffingOptionsIn.Builder builder, String key, boolean value) {
+    switch (key) {
+      case Content:
+        return builder.withContent(value);
+      case Dimensions:
+        return builder.withDimensions(value);
+      case Position:
+        return builder.withPosition(value);
+      case Structure:
+        return builder.withStructure(value);
+      case Style:
+        return builder.withStyle(value);
+      case Visual:
+        return builder.withVisual(value);
+    }
+    return builder;
+  }
+
+  private DiffingOptionsIn generateDiffingOptions(
+      List<String> enableOnly, List<String> disableOnly) {
+    if (enableOnly != null && !enableOnly.isEmpty()) {
+      DiffingOptionsIn.Builder builder = DiffingOptionsIn.builder();
+      for (String option : DiffingOptionValues) {
+        setDiffingOptionValue(builder, option, enableOnly.contains(option));
+      }
```

Comment:
Why not just loop through `enableOnly`?

Also, if you check if both, `enableOnly` and `disableOnly` are null, you can return early. That way the builder can be shared between disableOnly and enableOnly. Saves 2 lines.

Reply:
If I just loop with `enableOnly`, the extraneous values will be `null` not `false`.